esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -2245,6 +2245,46 @@ void WsWuInfo::getWUProcessLogSpecs(const char* processName, const char* logSpec
         logSpecs.sortAscii(false); //Sort the logSpecs from old to new
 }
 
+bool WsWuInfo::validateWUProcessLog(const char* file, bool eclAgent)
+{
+    Owned<IStringIterator> logs = cw->getLogs(eclAgent ? "EclAgent" : "Thor");
+    ForEach (*logs)
+    {
+        SCMStringBuffer logName;
+        logs->str(logName);
+        if (logName.length() < 1)
+            continue;
+
+        if (strieq(file, logName.str()))
+            return true;
+    }
+    return false;
+}
+
+bool WsWuInfo::validateWUAssociatedFile(const char* file, WUFileType type)
+{
+    Owned<IConstWUQuery> query = cw->getQuery();
+    if (!query)
+        return false;
+
+    Owned<IConstWUAssociatedFileIterator> iter = &query->getAssociatedFiles();
+    ForEach(*iter)
```

Comment:
future/efficiency: if there are a lot of associated files, and a lot of calls to validateWUAssociatedFile, this linear scanning could be a bit expensive. It's probably 'ok' because esp will be reopening the workunit in general frequently, so caching/a more efficient mechanism won't be available.